internal/resources/network_peer.go

```diff
@@ -0,0 +1,418 @@
+	)
+	cfg := api.EndpointCfg{Url: url, Method: http.MethodPost, SuccessStatus: http.StatusCreated}
+
+	response, err := n.Client.ExecuteWithRetry(
+		ctx,
+		cfg,
+		networkPeerRequest,
+		n.Token,
+		nil,
+	)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating network peer",
+			errorMessageWhileNetworkPeerCreation+api.ParseError(err),
+		)
+		return
+	}
+
+	networkPeerResponse := network_peer_api.GetNetworkPeeringRecordResponse{}
+	err = json.Unmarshal(response.Body, &networkPeerResponse)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating network peer",
+			errorMessageWhileNetworkPeerCreation+"error during unmarshalling:"+err.Error(),
+		)
+		return
+	}
+
+	diags = resp.State.Set(ctx, initializeNetworkPeerPlanId(plan, networkPeerResponse.Id.String()))
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	refreshedState, err := n.retrieveNetworkPeer(ctx, organizationId, projectId, clusterId, networkPeerResponse.Id.String(), plan.ProviderType.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error reading network peering service status",
+			"Error reading network peering service status, unexpected error: "+err.Error(),
+		)
+
+		return
+	}
+
+	refreshedState.ProviderType = plan.ProviderType
+
+	diags = resp.State.Set(ctx, refreshedState)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func (n *NetworkPeer) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state providerschema.NetworkPeer
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	IDs, err := state.Validate()
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error Reading Private Endpoint",
+			"Could not validate private endpoint "+state.Id.String()+": "+err.Error(),
+		)
+		return
+	}
+
+	var (
+		organizationId = IDs[providerschema.OrganizationId]
+		projectId      = IDs[providerschema.ProjectId]
+		clusterId      = IDs[providerschema.ClusterId]
+		peerId         = IDs[providerschema.Id]
+	)
+
+	refreshedState, err := n.retrieveNetworkPeer(ctx, organizationId, projectId, clusterId, peerId, state.ProviderType.ValueString())
+	if err != nil {
+		resourceNotFound, errString := api.CheckResourceNotFoundError(err)
+		if resourceNotFound {
+			tflog.Info(ctx, "resource doesn't exist in remote server removing resource from state file")
+			resp.State.RemoveResource(ctx)
+			return
+		}
+		resp.Diagnostics.AddError(
+			"Error reading private endpoint status",
+			"Error reading private endpoint status, unexpected error: "+errString,
+		)
+
+		return
+	}
+
+	diags = resp.State.Set(ctx, &refreshedState)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Update there is no update API so returns an error.
+func (n *NetworkPeer) Update(_ context.Context, _ resource.UpdateRequest, resp *resource.UpdateResponse) {
```

Comment:
As Aniket correctly pointed out in another comment, use `requireReplace`.  This means no need to throw an error with update.  Just leave it blank.  See https://developer.hashicorp.com/terraform/plugin/framework/resources/update#caveats:

```
If the resource does not support modification and should always be recreated on configuration value updates, the Update logic can be left empty and ensure all configurable schema attributes implement the [resource.RequiresReplace() attribute plan modifier](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/resource#RequiresReplace).
```

Reply:
Ok cool, I have removed the error and added the godoc for the explanation. 👍